### PR TITLE
autohostname: grudgingly accept the use of fqdns as short hostnames

### PR DESCRIPTION
**Pull request by fmap**

### CODE REVIEW

File: nixos/ec2/autohostname.nix [hunk 1]
```diff
@@ -44,7 +45,7 @@ let
 
     signature=$(echo -n $date | ${openssl} dgst -binary -sha1 -hmac $1 | ${base64})
     auth_header="X-Amzn-Authorization: AWS3-HTTPS AWSAccessKeyId=$2,Algorithm=HmacSHA1,Signature=$signature"
-    hostname=$(${hostname}).${zone}
+    hostname=$(${hostname}); ${fgrep} -q ${zone} <<< $hostname || hostname=$hostname.${zone}
```
Comment (proger) [inline nixos/ec2/autohostname.nix]: what about just setting system's hostname to contain the whole FQDN instead?

Comment (ip1981) [inline nixos/ec2/autohostname.nix]: I wish it were so.

Comment (proger) [inline nixos/ec2/autohostname.nix]: I mean here: https://github.com/zalora/microgram/blob/fqdn-lol/nixos/ec2/autohostname.nix#L94

@ip1981 why not?

Comment (ip1981) [inline nixos/ec2/autohostname.nix]: Sorry, I mean I'd like FQDN :-)

Comment (fmap) [inline nixos/ec2/autohostname.nix]: I can live with that.

**Comment by proger** [comment]
:+1: LGTM

**Comment by phunehehe** [comment]
:+1:

**Comment by fmap** [comment]
Don't merge this yet. For one thing, it's broken: `zone` is out of scope in the `hostname` invocation. And the semantics of `route53RegisterHostname` are such that one instance can have multiple domain names and zones (which makes sense if just for `int`), and I'm not yet sure how we should select between them (if this is still what we want at all.)

**Comment by fmap** [comment]
https://github.com/zalora/microgram/commit/d948c7036cf3a1edeaecae87b7abb26c397a32ec  + https://github.com/zalora/eris/commit/d03367c74a082867a5f7f9322a8bf30f76d0f334

**Pull request closed without merging**